toolchain/lexer/tokenized_buffer.cpp

```diff
@@ -907,33 +905,26 @@ auto TokenizedBuffer::SourceBufferLocationTranslator::GetLocation(
   int line_number = line_it - buffer_->line_infos_.begin();
   int column_number = offset - line_it->start;
 
-  llvm::StringRef line;
-
-  // We might still be lexing the last line. If so, check to see if there are
-  // any newline characters between the position we've finished lexing up to
-  // and the given location.
-  if (incomplete_line_info && column_number > *last_line_lexed_to_column_) {
-    column_number = *last_line_lexed_to_column_;
-    int64_t start = line_it->start;
-    for (int64_t i = line_it->start + *last_line_lexed_to_column_; i != offset;
-         ++i) {
-      if (buffer_->source_->text()[i] == '\n') {
-        start = i;
-        ++line_number;
-        column_number = 0;
-      } else {
-        ++column_number;
-      }
+  // Start by grabbing the buffer. If the line isn't fully lexed, the length
+  // will be npos and the buffer will be grabbed from the start; we'll then
+  // adjust the length.
+  llvm::StringRef line =
+      buffer_->source_->text().substr(line_it->start, line_it->length);
+  if (line_it->length == static_cast<int32_t>(llvm::StringRef::npos)) {
+    CARBON_CHECK(column_number < *last_line_lexed_to_column_ ||
```

Comment:
It makes me a little uncomfortable to read from `last_line_lexed_to_column_` here without checking that it's not null. (It's only non-null while lexing.) We shouldn't have any lines with a length of `-1` once we finish lexing, so we could `CHECK` this, if we want to stop using it to determine whether we're in an incomplete line.